src/Microsoft.VisualStudio.ProjectSystem.CSharp.VS/ProjectSystem/VS/CSharpProjectGuidProvider.cs

```diff
@@ -19,7 +19,6 @@ internal class CSharpProjectGuidProvider : IItemTypeGuidProvider
         [ImportingConstructor]
         public CSharpProjectGuidProvider(UnconfiguredProject unconfiguredProject)
```

Comment:
:bulb: You can improve the resistance of this type to incorrect use by applying one of the following approaches in addition to the change already made:

1. Mark the constructor `private` instead of `public`
2. Use `[Obsolete("This type may only be obtained through the MEF container", error: true)]`

Reply:
We use the constructors in tests.